singularity-eos/eos/eos_davis.cpp

```diff
@@ -139,28 +140,20 @@ PORTABLE_FUNCTION void DavisReactants::DensityEnergyFromPressureTemperature(
     const Real press, const Real temp, Real *lambda, Real &rho, Real &sie) const {
   // First, solve P=P(rho,T) for rho.  Note P(rho,e) has an sie-es term, which is only a
   // function of T
-  auto residual = [&](const Real r) {
-    return press - (Ps(r) + Gamma(r) * r * _Cv0 * Ts(r) / (1 + _alpha) *
-                                (std::pow(temp / Ts(r), 1 + _alpha) - 1.0));
+  auto PofRatT = [&](const Real r) {
+    return (Ps(r) + Gamma(r) * r * _Cv0 * Ts(r) / (1 + _alpha) *
+                        (std::pow(temp / Ts(r), 1 + _alpha) - 1.0));
   };
-  Real rho1 = _rho0, res1 = residual(rho1);
-  Real slope =
-      _Cv0 * _G0 *
-      ((_alpha * _G0 - 1.0) * temp * std::pow(temp / _T0, _alpha) + _T0 + _G0 * _T0) /
-      (1 + _alpha);
-  Real rho2 = rho1 + res1 / slope, res2 = residual(rho2);
-  Real rhom, resm;
-  for (int i = 1; i < 20; ++i) {
-    slope = (rho2 - rho1) / (res2 - res1);
-    rhom = rho2 - res2 * slope;
-    resm = residual(rhom);
-    if (resm / press < 1e-8) break;
-    rho1 = rho2;
-    res1 = res2;
-    rho2 = rhom;
-    res2 = resm;
+  using RootFinding1D::regula_falsi;
+  using RootFinding1D::Status;
+  RootFinding1D::RootCounts counts;
+  auto status =
+      regula_falsi(PofRatT, press, _rho0, 1.0e-5, 1.0e3, 1.0e-8, 1.0e-8, rho, counts);
```

Comment:
Is there a need to have better bounding guesses than hardcoded ones? Should we base them off of rho0?

Reply:
This is a good point. We could make an issue for that and address it later, though, when we're actually working more closely with davis.

Reply:
Possibly.  I know that the Regula Falsi solver gave me some trouble with wider bounds on a Gruneisen EOS (1e-12 -- 1e12) however, with an initial guess of `_rho0` across the board, we're likely to find the solution we desire.  There is logic (that pre-existed this MR) in the JWL version that checks to see if the incoming rho is set (which itself isn't a guarantee of anything good either).

I'll re-echo my previous comments that this sort of routine is likely limited to initialization and therefore initial-like states.  If that turns out to be insufficient, we could re-visit.

Reply:
Issue #158. Let's put a pin in it and re-visit when we're working more closely with this EOS.